packages/shared/src/hooks/useConditionalFeature.ts

```diff
@@ -0,0 +1,36 @@
+import { useQuery } from '@tanstack/react-query';
+import { JSONValue, WidenPrimitives } from '@growthbook/growthbook';
+import { useContext } from 'react';
+import { generateQueryKey, RequestKey } from '../lib/query';
+import { Feature } from '../lib/featureManagement';
+import { disabledRefetch } from '../lib/func';
+import AuthContext from '../contexts/AuthContext';
+import { useGrowthBookContext } from '../components/GrowthBookProvider';
+
+export const useConditionalFeature = <T extends JSONValue>({
+  feature,
+  shouldEvaluate,
+}: {
+  feature: Feature<T>;
+  shouldEvaluate: boolean;
+}): WidenPrimitives<T> => {
+  const { user } = useContext(AuthContext);
+  const { growthbook } = useGrowthBookContext();
+  const queryKey = generateQueryKey(RequestKey.Feature, user, feature.id);
+
+  const { data: featureValue } = useQuery(
+    queryKey,
+    () => {
+      return growthbook
```

Comment:
In case the gb object updates, shouldn't we retrigger this?

Reply:
I guess yes, not sure the current useFeatureValue does it.
But can add it in querykey somehow

Reply:
I'm actually not sure we need to retrigger?
Wouldn't it just reflect a next evaluation on it's own?

I think pretty much how everything works now (and why sometimes you need to hard refresh on apps)

Reply:
ok so let's leave it as it is. but just double check that it works this way